### DISCUSSION TITLE
[Feature Request]: Environment Telemetry: Support SDI-12 Sensors

### DISCUSSION THREAD

**Discussion by ndoo** [meshtastic/firmware#5465] [Ideas]
### Platform

Cross-Platform

### Description

Putting this out there in case anyone else in the community may have a use case for natively supporting SDI-12 sensors in Meshtastic.

https://github.com/EnviroDIY/Arduino-SDI-12

**Comment by thebentern** [comment]
Probably a use case for integrating with the serial module as all of our Environment sensors have a hard requirement of being i2c, except for built in ones in the case of the T1000E

**Comment by garthvh** [comment]
We are also running low on sensor library space so this is a perfect item to use via the linked Arduino library and serial module.

**Reply by ndoo** [reply]
Hi there, I suppose you mean to send serial data natively over the mesh to process it elsewhere by way of decoding the protobuf wherever I need the data?

Yes that seems like a straightfoward solution, however we do lose the ability to natively view the telemetry on a Meshtastic device such as the RAK WisMesh Tap or the Seeed SenseCAP Indicator.

But I do understand that given the space constraints and being mindful of feature creep, perhaps this isn't something that should be integrated. My 2 cents though, is that SDI-12 and ModBus/RS-485 sensors do seem popular in the industrial/agriculture space.
